File: Output Algorithms/Custom/RGB/Micro:Bit.py
from microbit import *

animation = [...]

def plot(x, y, r, g, b):
    # Insert your code here

def clear():
    # Insert your code here

while True:
    if button_a.is_pressed():
        for i in range(len(animation)):
            bits = []
            for j in range(width * height):
                bits.append([])
                bits[j].append(animation[i][j] >> 16 & 255)
                bits[j].append(animation[i][j] >> 8 & 255)
                bits[j].append(animation[i][j] & 255)
            for j in range(width * height):
                x = j % width
                y = j // width
                plot(x, y, bits[j][0], bits[j][1], bits[j][2])
            sleep(1000 / fps)
            clear()

File: Output Algorithms/Custom/RGB/Micro:Bit.ts
const animation: number[][] = [...];

const plot = (x: number, y: number, r: number, g: number, b: number) => {
    // Insert your code here
};

const clear = () => {
    // Insert your code here
};

input.onButtonPressed(Button.A, () => {
    for (let i = 0; i < animation.length; ++i) {
        let bits: number[][] = [];
        for (let j = 0; j < width * height; ++j) {
            bits[j] = [];
            bits[j][0] = animation[i][j] >> 16 & 255;
            bits[j][1] = animation[i][j] >> 8 & 255;
            bits[j][2] = animation[i][j] & 255;
        }
        for (let j = 0; j < width * height; ++j) {
            const x: number = j % width;
            const y: number = Math.floor(j / width);
            plot(x, y, bits[j][0], bits[j][1], bits[j][2]);
        }
        basic.pause(1000 / fps);
        clear();
    }
});